libbeat/outputs/logstash/protocol.go

```diff
@@ -285,6 +295,49 @@ func (p *protocol) serializeDataFrame(
 	return nil
 }
 
+func serializeEvent(event common.MapStr, beat []byte) ([]byte, error) {
```

Comment:
it's tested (block-box testing sendEvents), as we've got some logstash unit tests forwarding events via sendEvents with server part decoding json messages for validation.